samples/shell/README.rst

```diff
@@ -0,0 +1,144 @@
+.. _testing_shell_sample:
+
+Testing shell
+#############
+
+Overview
+********
+
+This sample demonstrates how a pytest test can be created based on an already existing Zephyr application:
+a sample for the `shell module <https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/subsys/shell/shell_module>`_.
+It also presents how a pytest test can communicate with an application through DUT's (device under test) interfaces and verify it's output.
+
+**This example is not ment to be a torough test of the shell sample**.
+
+Building and Running
+********************
+
+`A checkouted Zephyr repository is needed to execute this sample.`
+
+#. Copy the source file of the test (`shell_test.py`) into a directory with the shell module sample (`zephyr/samples/subsys/shell/shell_module`).
```

Comment:
you can also create `testspec.yaml` in `shell_test.py` directory, and point a `sourc_dir` (absolute path or erlated to ZEPHYR_BASE), e.g.
```
common:
  source_dir: samples/subsys/shell/shell_module
tests:
  sample.shell.shell_module:
     ...
```

Reply:
Added